amida-tech/indaba-client
# Bug/inba 975 weighted value row delete error

#### What does this PR do?

Fixes an issue where deleting a weighted value row was broken. For example, this is before I delete the middle row:
![image](https://user-images.githubusercontent.com/4026454/52450692-2613a600-2b0a-11e9-8c1f-fc8bab0393fe.png)

And this is after I delete the middle row (weighted value 3):
![image](https://user-images.githubusercontent.com/4026454/52450714-3c216680-2b0a-11e9-9add-6d8487fc7dff.png)

Note that "Text three" moved up but the weighted value was "3" instead of "4", which it should have been. 

#### Related JIRA tickets:
INBA-975

#### How should this be manually tested?

1. Go into survey builder.
2. Selected a question with a weighted value (Checkbox, Multiple Choice, Dropdown).
3. You can check with one row, but you should check with multiple rows. 
4. Click "Weighted Value Question"
5. Enter values into each of the boxes that appear. Make these values different (ie 1 thru 10) so you can see the difference.
6. Delete a row at the top or midstream. The one below it should move up and the weighted value it was assigned should replace it:

Also, please try to break a "scale" question, which was slightly refactored. Minimum values SHOULD NOT go over a maximum value (instead being reset to one less than the max). 

#### Background/Context

#### Screenshots (if appropriate):
